server/session/handler_inventory_transaction.go

```diff
@@ -16,6 +16,10 @@ type InventoryTransactionHandler struct{}
 func (h *InventoryTransactionHandler) Handle(p packet.Packet, s *Session) error {
 	pk := p.(*packet.InventoryTransaction)
 
+	if len(pk.LegacySetItemSlots) > 0 {
+		h.resendInventories(s)
```

Comment:
As discussed on discord, please change this to call resendInventories for offhand only and then send the specific slot for anything else